### ISSUE TITLE
schema modules should export schemas

### ISSUE THREAD

**Issue by kjprice** [chapinkapa/Bold#8]
One thing that I came across is that it appears the we have [Bold.schemas](https://github.com/chapinkapa/Bold/blob/master/lib/bold.js#L247) actually expecting to hold instances of [mongo models](https://github.com/chapinkapa/Bold/blob/master/example/config/schema/user-store.js#L10).

It might make sense to create another directory *config/model* that will simply create models from the given schemas.

### config/model/user.js example

```
var Connection = require('bold').Connection;
var userSchema = require('../schema/user');

module.exports = Connection.model('User', userSchema);
```

### config/schema/user.js example

```
var mongoose = require('mongoose');

var User = new Schema({});

module.exports = User;
```

This becomes more important later when we need to access the UserSchema.

**Comment by chapinkapa** [comment]
Why will we need to access the UserSchema? I am pretty sure the UserSchema is assessable from the user  model.

**Comment by kjprice** [comment]
The problem that I found was that my UserSchema contained an array of another schema type (say DocumentSchema). What I tried to do was something like this:

```
var User = new Schema({
   firstName: String,
   lastName: String,
   documents: [DocumentSchema]
});
```

I was hoping that I could then access `DocumentSchema` using `Bold.Schemas.DocumentSchema`. However, `Bold.Schemas.DocumentSchema` is actually the model for `DocumentSchema`. That syntax was confusing and it took me a while to figure out what was going. I'm not sure how to access a schema from a model, but that does not solve the issue of the fact that `Bold.Schemas` is a list of models, which is confusing.

Another thing that we could do is have the `schema/user.js` file look like this at the end:

```
module.exports.schema = userSchema;
module.exports.model = Connection.model('User', userSchema);
```

Then Bold could create `Bold.Schemas` and `Bold.Models` after this.

Another idea would be just to get rid of `Bold.Schemas` altogether and just use `Bold.Models` and access the schemas through the model. This would be less confusing but would still be a hassle when a user wants to access the schemas.
